lib/ansible/modules/cloud/digital_ocean/do_floating_ip.py

```diff
@@ -0,0 +1,329 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# (c) 2015, Patrick F. Marques <patrickfmarques@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'metadata_version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: do_floating_ip
+short_description: Manage DigitalOcean Floating IPs
+description:
+     - Create/delete/assign a floating IP.
+version_added: "2.4"
+author: "Patrick Marques (@patrickfmarques)"
+options:
+  state:
+    description:
+     - Indicate desired state of the target.
+    default: present
+    choices: ['present', 'absent']
+  ip:
+    description:
+     - Public IP address of the Floating IP. Used to remove an IP
+    required: false
+    default: None
+  region:
+    description:
+     - The region that the Floating IP is reserved to.
+    required: false
+    default: None
+  droplet_id:
+    description:
+     - The Droplet that the Floating IP has been assigned to.
+    required: false
+    default: None
+  oauth_token:
+    description:
+     - DigitalOcean OAuth token.
+    required: true
+
+notes:
+  - Version 2 of DigitalOcean API is used.
+requirements:
+  - "python >= 2.6"
+'''
+
+
+EXAMPLES = '''
```

Comment:
Good points, the first I completely agree (I was using the env var and I forgot to add that one).
About the name of the **_token*. If you are asking my opinion, I called it *ouath* because is what it is although, I'm fine with keeping the old name or at least make an alias to keep backward compatibility (if this makes sense).